### PR TITLE
improv(commons): expand type utils functions

### PR DESCRIPTION
**Pull request by dreamorosi**
### Related issues, RFCs

**Issue number:** #2188

## Checklist

- [x] [My changes meet the tenets criteria](https://docs.powertools.aws.dev/lambda/typescript/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [ ] I have made corresponding changes to the *documentation*
- [x] My changes generate *no new warnings*
- [ ] I have *added tests* that prove my change is effective and works
- [x] The PR title follows the [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

***Is it a breaking change?:*** NO

- [ ] I have documented the migration process
- [ ] I have added, implemented necessary warnings (if it can live side by side)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: packages/commons/src/typeUtils.ts [hunk 1]
```diff
@@ -0,0 +1,177 @@
+/**
+ * Returns true if the passed value is a record (object).
+ *
+ * @param value The value to check
+ */
+const isRecord = (value: unknown): value is Record<string, unknown> => {
```
Comment (am29d) [inline packages/commons/src/typeUtils.ts:6]: the keys in the `Record` doesn't have to strings only, afaik we can have `string | number | symbol`

File: packages/commons/src/typeUtils.ts [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+const isIntegerNumber = (value: unknown): value is number => {
+  return isNumber(value) && Number.isInteger(value);
+};
+
+/**
+ * Check if a value is truthy.
+ *
+ * @param value The value to check
+ */
+const isTruthy = (value: unknown): boolean => {
```
Comment (am29d) [inline packages/commons/src/typeUtils.ts:45]: `NaN` and functions would not pass here, but I am not sure we need them, just asking if left out this edge cases on purpose.

File: packages/commons/tests/unit/typeUtils.test.ts [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+        // Act
+        const result = isTruthy(value);
+
+        // Assess
+        expect(result).toBe(true);
+      }
+    );
+
+    it.each(['', 0, false, [], {}, Symbol])(
+      'returns true when the passed value is falsy',
```
Comment (am29d) [inline packages/commons/tests/unit/typeUtils.test.ts:64]:
```suggestion
      'returns false when the passed value is falsy',
```

File: packages/commons/src/typeUtils.ts [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+  }
+};
+
+/**
+ * Check if a value is null.
+ *
+ * @param value The value to check
+ */
+const isNull = (value: unknown): value is null => {
+  return Object.is(value, null);
```
Comment (am29d) [inline packages/commons/src/typeUtils.ts:67]: what is the difference between `value === null` and `Object.is`? It looks like they yield same result, is it rather a preference?

**Review by am29d — CHANGES_REQUESTED** [review]
Nice packaged PR, just a minor question on the `NaN` edge case.

File: packages/commons/src/typeUtils.ts (hunk 2 above)
Comment (dreamorosi) [reply packages/commons/src/typeUtils.ts:45]: True.

I don't think we need to include them as they're not JSON serializable anyway.

If in the future we need to check truthiness of these values we can still expand this function.

File: packages/commons/src/typeUtils.ts (hunk 4 above)
Comment (dreamorosi) [reply packages/commons/src/typeUtils.ts:67]: In the context of `null` the two notations are exactly equivalent, there are [other cases](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is#description) where `Object.is` is preferable but this is not one of them.

In this case it was just preference. In my mind it _reads_ a bit more clear since it's closer to plain English.

File: packages/commons/src/typeUtils.ts (hunk 2 above)
Comment (am29d) [reply packages/commons/src/typeUtils.ts:45]: > they're not JSON serializable anyway. 

makes sense

**Review by am29d — DISMISSED** [review]

**Review by am29d — DISMISSED** [review]
